src/components/Home.jsx

```diff
@@ -17,15 +17,27 @@ const Home = () => {
 
     db.collection('tokens')
       .doc('tokenDocument')
-      .update({
-        tokenArray: firebase.firestore.FieldValue.arrayUnion(token),
-      })
-      .catch((error) => {
-        db.collection('tokens')
-          .doc('tokenDocument')
-          .set({
-            tokenArray: [token],
-          });
+      .get()
+      .then((doc) => {
+        if (doc.exists) {
+          db.collection('tokens')
+            .doc('tokenDocument')
+            .update({
+              tokenArray: firebase.firestore.FieldValue.arrayUnion(token),
+            })
+            .catch((error) => {
+              console.log(error);
+            });
+        } else {
+          db.collection('tokens')
+            .doc('tokenDocument')
+            .set({
+              tokenArray: [token],
+            })
+            .catch((error) => {
+              console.log(error);
+            });
```

Comment:
Consider returning in each of these catch blocks, otherwise it will fall through to the redirect, which will probably push back to home. We can prevent that unnecessary redirect by returning early.

Reply:
Thank you, I hadn't thought of this ☺️